source/predicates/array.ts

```diff
@@ -158,4 +160,29 @@ export class ArrayPredicate<T = unknown> extends Predicate<T[]> {
 			}
 		});
 	}
+
+	/**
+	Test an array to match the `predicates` exactly.
+
+	The predicates is an array which describes how the tested array should look like. Each predicate is the same as the source array.
+
+	@param predicates - The predicate placed at the nth index should be applied also to the item at the same index.
+
+	@example
+	```
+	ow(['1', 2], ow.array.exactShape([ow.string, ow.number]));
+	```
+	*/
+	exactShape(predicates: Predicate[]) {
+		const shape: Shape = {};
+
+		for (const [index, predicate] of predicates) {
+			shape[index] = predicates;
+		}
```

Comment:
Couldn't this simply be `const shape = predicates as Shape`?

Arrays in JS are objects.